src/tools/illink/src/ILLink.RoslynAnalyzer/TrimAnalysis/GenericArgumentDataFlow.cs

```diff
@@ -21,12 +21,20 @@ public static void ProcessGenericArgumentDataFlow (Location location, IMethodSym
 		{
 			ProcessGenericArgumentDataFlow (location, method.TypeArguments, method.TypeParameters, reportDiagnostic);
 
-			ProcessGenericArgumentDataFlow (location, method.ContainingType, reportDiagnostic);
+			var type = method.ContainingType;
+			while (type is { IsGenericType: true }) {
```

Comment:
Same as below
```suggestion
			while (type.IsGenericType) {
```

Reply:
Also, would it make sense to recurse up the `ContainingType`s in the `INamedTypeSymbol` overload of `ProcessGenericArgumentDataflow` rather than in both the `IMethodSymbol` and `IFieldSymbol` overloads, or does doing that cause issues elsewhere?